lib/utilities.ak

```diff
@@ -71,19 +107,39 @@ pub fn payout_sum(
           // The datum must be well formed
           expect InlineDatum(datum) = output.datum
           expect datum: VendorDatum = datum
-          let this_payout_sum =
+          let (this_payout_sum, payout_count) =
             datum.payouts
               |> list.foldl(
-                  assets.zero,
+                  (assets.zero, 0),
                   fn(payout, acc) {
+                    let value = assets.from_asset_list(payout.value)
```

Comment:
NIT: Maybe a comment reminding that `from_asset_list` is enforcing some invariants for us could be useful for future reference. In particular, it throws on duplicate policies in the list, which could be used to circumvent the limitations on datum size we are trying to enforce below